disease_gene/disease_associates_gene/generative_model_experiments/gen_model_benchmarking.py

```diff
@@ -0,0 +1,335 @@
+        grid_results[str(param)] = label_model.predict_proba(model_data[1])[:,0]
+        
+    model_grid_search[model_label] = pd.DataFrame.from_dict(grid_results)
+
+
+# In[16]:
+
+
+model_grid_aucs = {}
+for model in model_grid_search:
+    model_grid_aucs[model] = plot_curve(model_grid_search[model], candidate_dfs['dev'].curated_dsh, 
+                               figsize=(16,6), model_type='scatterplot', plot_title=model, metric="ROC", font_size=10)
+
+
+# In[17]:
+
+
+model_grid_auc_dfs = {}
+for model in model_grid_aucs:
+    model_grid_auc_dfs[model] = (
+        get_auc_significant_stats(candidate_dfs['dev'], model_grid_aucs[model])
+        .sort_values('auroc', ascending=False)
+    )
+    print(model)
+    print(model_grid_auc_dfs[model].head(5))
+    print()
+
+
+# # Final Evaluation on Held out Hand Labeled Test Data
+
+# In[39]:
+
+
+dev_model_df = pd.DataFrame()
+for best_model, model_data, model_label in zip([1.083, 2.067, 1.575], validation_data, model_labels):
+    label_model = LabelModel(k=2, seed=100)
+    label_model.train_model(model_data[0] , n_epochs=1000, verbose=False, lr=0.01, l2=best_model)
+    dev_model_df[model_label] = label_model.predict_proba(model_data[1])[:,0]
+
+
+# In[40]:
+
+
+_ = plot_curve(
+    dev_model_df, 
+    candidate_dfs['dev'].curated_dsh,
+    model_type='curve', figsize=(10,8), 
+    plot_title="Disease Associates Gene AUROC on Dev Set", font_size=16
+)
+
+
+# In[43]:
+
+
+_ = plot_curve(
+    dev_model_df, 
+    candidate_dfs['dev'].curated_dsh,
+    model_type='curve', figsize=(12,7), 
+    plot_title="DaG Precision Recall Curve on Dev",
+    metric='PR', font_size=16
+)
+
+
+# In[21]:
+
+
+label_model = LabelModel(k=2, seed=100)
+label_model.train_model(validation_data[1][0], n_epochs=1000, verbose=False, lr=0.01, l2=2.067)
+dev_predictions = convert_labels(label_model.predict(validation_data[1][1]), 'categorical', 'onezero')
+dev_marginals = label_model.predict_proba(validation_data[1][1])[:,0]
+
+
+# In[22]:
+
+
+plt.rcParams.update({'font.size': 16})
+plt.figure(figsize=(10,6))
+plot_predictions_histogram(
+    dev_predictions,
+    candidate_dfs['dev'].curated_dsh.astype(int).values,
+    title="Prediction Histogram for Dev Set"
+)
+
+
+# In[23]:
+
+
+confusion_matrix(
+    convert_labels(candidate_dfs['dev'].curated_dsh.values, 'onezero', 'categorical'),
+    convert_labels(dev_predictions, 'onezero', 'categorical')
+)
+
+
+# In[25]:
+
+
+lf_summary(label_matricies['dev'], Y=candidate_dfs['dev'].curated_dsh.apply(lambda x: 1 if x > 0 else 2).values, lf_names=lf_names)
+
+
+# In[26]:
+
+
+plot_label_matrix_heatmap(convert_labels(label_matricies['dev'].toarray(), 'categorical', 'plusminus').T, 
+                          yaxis_tick_labels=lf_names, 
+                          figsize=(10,12), font_size=10)
+
+
+# In[27]:
+
+
+pickle.dump(label_model.predict_proba(L[:, :24]), open("data/train_marginals.pkl", "wb"))
+
```

Comment:
Might add the file name to the top of the script or make it an argument if it is going to change in the future